### DISCUSSION TITLE
Changes on file will still show in text editor after I changed to another branch

### DISCUSSION THREAD

**Discussion by hannarosenfeld** [community/community#40298] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hi, I have an issue with git branching.

When I change branches, the state of the file is not reflected in VSC. I tried this out in a different text editor (emacs) and it's the same there.

For example, when I branch of, make changes, and switch back to master, the text editor does not switch back to the state of master. The changes I made to the file on the other branch will persist in the text editor.

Does someone have a idea on how to fix this?

**Comment by airtower-luna** [comment, marked as answer]
Tell your text editor to reload the file. None of the editors I know does that automatically. Emacs does ask what to do if you type in an open file that has changed on disk. :wink:
